src/hooks/Auth.ts

```diff
@@ -97,84 +119,107 @@ export const useAuth = ({
       return;
     }
 
-    setToken(oResponse.accessToken);
-
     if (oResponse.refreshToken) {
-      setRefreshToken(oResponse.refreshToken);
+      setRefreshToken({ token: oResponse.refreshToken, issuedAt: new Date(), expiresIn: (oResponse.expiresIn || 3600) * 1000 });
       void storage.setItem(AUTH_REFRESH_TOKEN_KEY, oResponse.refreshToken);
     }
 
+    setToken(oResponse.accessToken);
     setIdToken(oResponse.idToken);
-    setIsAutoLoginDone(true);
-
-    // Set the interval to a bit shorter time.
-    setRefreshInterval(Math.floor((oResponse.expiresIn || 3600) * 1000 * 0.9));
   }, []);
 
+  const _refreshAccessToken = useCallback(
+    async (savedRefreshToken: string) => {
+      if (!configuration) {
+        throw new Error('called refresh too soon - you can check that with "isReady"');
+      }
+      try {
+        const response = await performRefreshTokenRequest(
+          configuration,
+          options.clientId,
+          options.redirectUrl,
+          savedRefreshToken,
+          options.tokenRequest?.extras,
+        );
+        setTokenResponse(response);
+        return { token: response.accessToken, idToken: response.idToken };
+      } catch (err) {
+        if (err instanceof AppAuthError) {
+          const statusCode = Number.parseInt(err.message, 10);
+          if (statusCode >= 400 && statusCode < 500) {
+            // HTTP client error -> token is probably expired
+            console.log('Removing expired refresh token');
```

Comment:
Libs should not log to console as the user has no control of it and it may just clutter the log and annoy the developer.
If you still want it, there could be a `debug` boolean to enable that.
I think there is one other console.log still in the code...